src/buttons/RolesEditionButton/RolesEditionButton.js

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Button, IconButton, Tooltip } from '@material-ui/core';
+import ShareIcon from '@material-ui/icons/Share';
+import RolesEditionDialog from './components/RolesEditionDialog/RolesEditionDialog';
+
+export const RolesEditionButton = ({
+  labels,
+  isIconButton,
+  users,
+  scenarioRolesPermissionsMapping,
+  isReadOnly,
+  onConfirmChanges,
+  specificRolesByAgent,
+  defaultAccess,
+  scenarioId,
+}) => {
+  const [open, setOpen] = useState(false);
+  const openDialog = () => setOpen(true);
+  const closeDialog = () => setOpen(false);
+  const buttonContent = isIconButton ? (
+    <IconButton data-cy="share-scenario-button" size="medium" variant="outlined" onClick={openDialog} color="primary">
+      <ShareIcon />
+    </IconButton>
+  ) : (
+    <Button data-cy="share-scenario-button" size="medium" variant="outlined" onClick={openDialog} color="primary">
+      {labels.button.title}
+    </Button>
+  );
+  return (
+    <div>
+      <Tooltip arrow title={labels.button.tooltip}>
+        <div>{buttonContent}</div>
+      </Tooltip>
+      <RolesEditionDialog
+        scenarioId={scenarioId}
+        open={open}
+        scenarioRolesPermissionsMapping={scenarioRolesPermissionsMapping}
+        specificRolesByAgent={specificRolesByAgent}
+        defaultAccess={defaultAccess}
+        users={users}
+        labels={labels.dialog}
+        isReadOnly={isReadOnly}
+        onConfirmChanges={onConfirmChanges}
+        closeDialog={closeDialog}
+      />
+    </div>
+  );
+};
+RolesEditionButton.propTypes = {
+  labels: PropTypes.shape({
+    button: PropTypes.shape({
+      title: PropTypes.string.isRequired,
```

Comment:
Title should probably not be _required_ if we provide the "icon only" mode for the button (we may have to hard-code a default value though: `labels?.button?.title ?? 'Share'`)

Reply:
Is a defaultProps enough (if there no title key, it returns `undefined` and normally uses the default prop as we learned last week :laughing: ) or it's better to add a condition in the component ?

Reply:
That would depend of whether or not `labels` and `labels.button` are required. It would probably be safer to add a fallback in the component :)